src/GenesisPanelView.cpp

```diff
@@ -1811,17 +1811,34 @@ void PanelView::Edit(CustomListItem *item)
 	{
 		case FT_FILE:
 		case FT_SYMLINKFILE:
-			BString file;
-			file.SetTo(m_Path.String());
-			file+="/";
-			file+=item->m_FileName;
+			BString app, file, command;
 
-			BString tempstring;
-			tempstring.SetTo("StyledEdit \"");
-			tempstring << file.String();
-			tempstring << "\" &";
-			system(tempstring.String());
+			app = SETTINGS->GetEditorApp();
+			if (app.IsEmpty())
+			{
+				BAlert *alert = new BAlert("Genesis","You have to choose editor app in the Preferences screen","Okay","Open Preferences",NULL,B_WIDTH_AS_USUAL,B_OFFSET_SPACING,B_STOP_ALERT);
```

Comment:
Suggestion, considering standard Haiku naming: Choose->Select (see FileTypes prefs e.g.), Okay->OK
```suggestion
				BAlert *alert = new BAlert("Genesis","You have to select an editor application in the Genesis preferences","OK","Open preferences",NULL,B_WIDTH_AS_USUAL,B_OFFSET_SPACING,B_STOP_ALERT);
```